### ISSUE TITLE
Implement earthkit xarray engine

### ISSUE THREAD

**Issue by sandorkertesz** [ecmwf/earthkit-data#400]
The task is to continue the developments in branch `feature/earthkit-engine` to develop a new xarray engine to replace `cfgrib` for grib -> xarray -> grib conversions.

**Comment by sandorkertesz** [comment]
TODO:
- [] ensure lazy loading, the current implementation loads the whole grib into memory at a point
- [] add tests
- [] compare speed with cfgrib
- [] add option to `to_xarray()` to choose between `earthkit` and `cfgrib` engines

**Comment by sandorkertesz** [comment]
Use case for benchmarking
----------------------------

 GRIB data containing 118 global 0.25x0.25 fields

```python
    fl = from_source("file", "pl_025.grib")
    xa = fl.to_xarray()
```
`mem` in the table below shows the memory increase during the `to_xarray()` call.

|engine   | time (s) | mem (MB)| peak mem (MB)|
|-------- |----   |---- |---|
|cfgrib   |  1.68 | 94| 275|
|earthkit | 4.08 |534|727|

**Comment by sandorkertesz** [comment]
The current implementation is using the "mars profile" based on the mars ecCodes keys. The idea is to implement other predefined profiles e.g. "CF" and allow users to define their own profiles. 

See example notebook: https://earthkit-data.readthedocs.io/en/feature-earthkit-engine/examples/xarray_engine_demo.html

**Comment by sandorkertesz** [comment]
Fixed by #473
